### PR TITLE
Add Omnibox suggestions and suggestion item sub components

### PR DESCRIPTION
**Pull request by josephschmitt**
Use the source function to show how to populate a list of suggestions. For its data source it's using Github's public [JSON file of emoji](https://api.github.com/emojis), and for a search engine it's using the [Fuse](http://fusejs.io/) project.

### CODE REVIEW

File: example/example.html [hunk 1]
```diff
@@ -1,13 +1,57 @@
...
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/2.5.0/fuse.min.js"></script>
     <script type="text/javascript">
-      angular.module('demoApp', ['ngc.omnibox']);
+      angular
+        .module('demoApp', ['ngc.omnibox'])
+        .controller('OmniboxExampleController', function ($http, $q, $scope) {
+          let fuse;
+
+          $http.get('https://api.github.com/emojis').then(function (response) {
+            var emoji = Object.keys(response.data).map(function (id) {
```
Comment (onlywei) [inline example/example.html]: You use `let` *and* `var` in the same file?

File: example/example.html [hunk 2]
```diff
@@ -1,13 +1,57 @@
...
+      <p>Model value: {{demo.model}}</p>
     </main>
 
     <script src="../dist/ngc-omnibox.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/2.5.0/fuse.min.js"></script>
     <script type="text/javascript">
-      angular.module('demoApp', ['ngc.omnibox']);
+      angular
+        .module('demoApp', ['ngc.omnibox'])
+        .controller('OmniboxExampleController', function ($http, $q, $scope) {
```
Comment (onlywei) [inline example/example.html]: Do you need to inject `$scope`?

File: example/example.html (hunk 1 above)
Comment (josephschmitt) [inline example/example.html]: Wait, I thought I fixed that. Might've been in the next PR.

File: example/example.html (hunk 2 above)
Comment (josephschmitt) [inline example/example.html]: @onlywei done

**Pull request merged**